src/atoms/Layout/index.js

```diff
@@ -0,0 +1,78 @@
+import React, { PropTypes } from 'react';
+import classnames from 'classnames';
+
+import styles from './layout.module.scss';
+
+function Layout ( props ) {
+  const {
+    children,
+    childCols,
+    breakpoint,
+    horizontalAlign,
+    verticalAlign
+  } = props;
+
+  let colsIdx = -1;
+
+  const kids = React.Children.map( children, ( child, idx ) => {
+    if ( !childCols[++colsIdx] ) {
+      colsIdx = 0;
+    }
+
+    return (
+      <div
+        className={ classnames(
+          styles[`column-${ childCols[colsIdx] || 12 }`]
```

Comment:
I still think that the logic here is too opaque. The code is really nice tho -- so, personally, I'd like to see some comments here explaining what's going on. I think it's just a nice thing to do for any future engineers that might need to come in here.